hnn_core/docs.py

```diff
@@ -0,0 +1,551 @@
+"""The documentation functions."""
+
+docdict = dict()
+# Define docdicts
+
+docdict[
+    "net"
+] = """
+net : Instance of Network object
+    The Network object.
+"""
+
+docdict[
+    "fname"
+] = """
+fname : str | Path object
+    Full path to the output file (.hdf5).
+"""
+
+docdict[
+    "overwrite"
+] = """
+overwrite : Boolean
+    True : Overwrite existing file.
+    False : Throw error if file already exists.
+"""
+
+docdict[
+    "save_unsimulated"
+] = """
+save_unsimulated : Boolean
+    True : Do not save the Network simulation output.
+    False : Save complete Network as provided in input.
+"""
+
+docdict[
+    "read_raw"
+] = """
+read_raw : Boolean
+    True : Read unsimulated network.
+    False : Read simulated network.
+"""
+
+docdict[
+    "L"
+] = """
+L : float
+    length of a section in microns.
+"""
+
+docdict[
+    "diam"
+] = """
+diam : float
+    diameter of a section in microns.
+"""
+
+docdict[
+    "cm"
+] = """
+cm : float
+    membrane capacitance in micro-Farads.
+"""
+
+docdict[
+    "Ra"
+] = """
+Ra : float
+    axial resistivity in ohm-cm.
+"""
+
+docdict[
+    "end_pts"
+] = """
+end_pts : list of [x, y, z]
+    The start and stop points of the section.
+"""
+
+docdict[
+    "syns"
+] = """
+syns : list of str
+    The synaptic mechanisms to add in this section.
+"""
+
+docdict[
+    "mechs"
+] = """
+mechs : dict
+    Mechanisms to insert in this section. The keys
+    are the names of the mechanisms and values
+    are the properties.
+"""
+
+docdict[
+    "cell_name"
+] = """
+cell_name : str
+    Name of the cell.
+"""
+
+docdict[
+    "pos"
+] = """
+pos : tuple
+    The (x, y, z) coordinates.
+"""
+
+docdict[
+    "sections"
+] = """
+sections : dict of Section
+    Dictionary with keys as section name.
+"""
```

Comment:
let's remove entries which are not used anywhere and keep the bare minimum. This PR has become too big and I worry about maintaining the code.